docs/architecture.md

```diff
@@ -0,0 +1,111 @@
+
+`start_with_config` in `neard/src/lib.rs` is the actual entry point and it starts all the actors.
+
+## Code Map
+
+This section contains some high-level overview of important crates and data structures.
+
+### `core/primitives`
+
+This crate contains most of the types that are shared across different crates.
+
+### `core/primitives-core`
+
+This crate contains types needed for runtime. 
+
+**Architecture Invariant**: this crate needs to be published separately since near-skd-rs depends on it.
+It cannot have any dependencies on other nearcore crates.
+
+### `core/store/trie`
+
+This directory contains the MPT state implementation.
+Note that we usually use `TrieUpdate` to interact with the state.
+
+### `chain/chain`
+
+This crate contains most of the chain logic (consensus, block processing, etc). 
+`ChainUpdate::process_block` is where most of the block processing logic happens.
+
+**Architecture Invariant**: interface between chain and runtime is defined by `RuntimeAdapter`.
+All invocations of runtime goes through `RuntimeAdapter`
+
+### `chain/chunks`
+
+This crate contains most of the sharding logic which includes chunk creation, distribution, and processing.
+`ShardsManager` is the main struct that orchestrates everything here.
+
+### `chain/client`
+
+This crate defines two important structs, `Client` and `ViewClient`.
+`Client` includes everything necessary for the chain (without network and runtime) to function and runs in a single thread.
+`ViewClient` is a "read-only" client that answers queries without interfering with the operations of `Client`.
+`ViewClient` runs in multiple threads.
+
+### `chain/network`
+
+This crate contains the entire implementation of the p2p network used by NEAR blockchain nodes.
+
+Two important structs here: `PeerManagerActor` and `Peer`. 
+Peer manager orchestrates all the communications from network to other components and from other components to network.
+`Peer` is responsible for low-level network communications from and to a given peer.
+Peer manager runs in one thread while each `Peer` runs in its own thread.
+
+**Architecture Invariant**: Network communicates to `Client` through `NetworkClientMessages` and to `ViewClient` through `NetworkViewClientMessages`.
+Conversely, `Client` and `ViewClient` communicates to network through `NetworkRequests`.
+
+### `chain/epoch_manager`
+
+This crate is responsible for determining validators and other epoch related information such as epoch id for each epoch.
+
+Note: `EpochManager` is constructed in `NightshadeRuntime` rather than in `Chain`, partially because we had this idea of making epoch manager a smart contract.
+
+### `chain/jsonrpc`
+
+This crate implements [JSON-RPC](https://www.jsonrpc.org/) API server to enable submission of new transactions and inspection of the blockchain data, the network state, and the node status.
+
+### `runtime/runtime`
+
+This crate contains the main entry point to runtime -- `Runtime::apply`. 
```

Comment:
After reading this document, I think I am still confused about the difference between `chain` and `runtime`. Perhaps, somewhere in the birdeye view, it'd be uselfull to discuss that?

Perhaps a "transaction lifetime" story will be useful here? As in, "what happens when a dapp submits a transaction over JSON RPC"?